doc/HD44780.rst

```diff
@@ -0,0 +1,197 @@
+  from luma.core.render import canvas
+  from PIL import Image, ImageDraw
+
+  serial = parallel(RS=7, E=8, PINS=[25,24,23,27])
+  device = hd44780(serial)
+
+  def progress_bar(width, height, percentage):
+    img = Image.new('1', (width, height))
+    drw = ImageDraw.Draw(img)
+    drw.rectangle((0, 0, width-1, height-1), fill='black', outline='white')
+    drw.rectangle((0, 0, width*percentage, height-1), fill='white', outline='white')
+    return img
+
+  progress = 0.25
+  fnt = device.get_font(0)
+  with canvas(device) as draw:
+    draw.text( (5,0), f'Installing {progress*100:.0f}%', font=fnt, fill='white')
+    draw.bitmap( (5,8), progress_bar(70, 8, progress), fill='white')
+
+.. image:: images/hd44780_progress.jpg
+
+There are a few of things that deserve highlighting in this code.
+
+* We have used the hd44780 classes get_font method to retrieve the internal
+  font used by the device.  This enables us to place exact replicas of the
+  characters within the font tables on the canvas.  When these characters are
+  displayed, because they are already normal characters, they do not consume any
+  of the customer character RAM.  The hd44780 class contains both the A00 and A02
+  font tables.  You should request the table that matches what is installed in
+  your display.
+
+* The progress bar is drawn using normal ImageDraw primitives in this case a
+  couple of calls to the rectangle method.
+
+* The size of the progress bar was carefully chosen.  It is 70 pixels wide by 8
+  pixels high.  This will fill 14 characters worth of space.  Normally this
+  would be a problem.  However, the progress bar only requires four custom
+  characters regardless of what position the progress value is set to.  These
+  are:
+  - First character position filling with empty middle and end positions
+  - Filled first position with one filling middle position, other empty middle
+  positions and the empty end position
+  - Filled first position, filled middle positions and filling (or filled) end
+  position
+
+* The progress bar was carefully placed to align with the character cell
+  boundaries.  If we had placed the progress bar image one pixel to the left
+  if would have consumed two additional custom characters as the beginning and
+  end of the progress bar would be spread across character cell boundaries.  In
+  this particular case, it would still have displayed correctly because we would
+  be under the 8 character limit.
+
+
+Embedded Font Tables
+^^^^^^^^^^^^^^^^^^^^
+The :py:class:`luma.lcd.device.hd44780` class leverages
+:py:mod:`luma.core.bitmap_font` to include two fonts that replicate the two font
+tables that are commonly available for the hd44780.  These are:
+
+===========  ==================
+Font Table   Font Name
+===========  ==================
+A00          English Japanese
+A02          English European
+===========  ==================
+
+You can retrieve either of these fonts using the `get_font` method.
```

Comment:
use double backticks (or api link) instead of single

Reply:
Changed to double backticks